### PR TITLE
build: set up GitHub app PRLint

### PR DESCRIPTION
**Pull request by mistercrunch**
~~Sets up [gitlint](https://jorisroovers.com/gitlint/), which enforces [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and sets up the proper post commit hook to enforce it.~~

~~Also modifying the PR template to remove redundant annoying checkbox section. Once we adopt, I'm hoping we can modify our bot to auto-label based on the commit type.~~

~~This requires a bit more discipline upfront, but should lead to a cleaner `git log` in master, meaning better release notes and such.~~

- [x] waiting for Apache infra to install PRLint Github App. [JIRA](https://issues.apache.org/jira/browse/INFRA-20236) 

Sets up [PRLint](https://github.com/apps/prlint), with a single rule matching [Conventional
Commits](https://www.conventionalcommits.org/en/v1.0.0/)

Also modifying the PR template to remove redundant annoying checkbox section.

Once we adopt, I'm hoping we can modify our bot to auto-label based on the commit type.

### CODE REVIEW

**Comment by ktmud** [comment]
FYI, `superset-ui` actually [disabled commit lint](https://github.com/apache-superset/superset-ui/pull/363) recently, because we felt it's not really necessary given that we already had [Semantic Pull Request](https://github.com/zeke/semantic-pull-requests) check and will almost always do squash merge anyway.

Considering `incubator-superset` also only do squash merge, if this is to be added, would it be possible to enforce the checks only on the `master` branch?

**Comment by craig-rueda** [comment]
Agree. I'm all for making the commit log more readable. Typically, when squashing, the commits that are actually being squashed are appended into the squash commit's message. Let's give it a shot...

**Comment by nytai** [comment]
I agree with @ktmud a per check would be preferable. We can even enforce semantic commits on the actual PR. A local pre-commit check seems like it could get annoying really quickly. I usually fixup my commits or reword them before opening a pr or when submitting pr for final review, especially given the commit messages get squashed into the commit that's finally merged. However, I commit often when developing locally for things like moving branches or commits I intend to fixup. Having to write a message that'll pass commit lint, not to mention running commit lint, when I just want to save my work and move to another branch seems tedious.

**Comment by mistercrunch** [comment]
I agree local post-commit hook is heavy, there's a way to add a regex to skip something like `wip.*`. 

It's true that ultimately with that "squash & merge" github policy it's the PR title that matters most. Maybe we force a PR title check to merge instead?

**Comment by mistercrunch** [comment]
Oh! how about this: https://github.com/apps/prlint !?

**Comment by ktmud** [comment]
> Oh! how about this: [github.com/apps/prlint](https://github.com/apps/prlint) !?

Looks legit! Seems more powerful than [Semantic Pull Requests](https://github.com/zeke/semantic-pull-requests) that `superset-ui` is using.

**Comment by dpgaspar** [comment]
PRLint Looks good. Something like this would really help on patch releases and making the changelog more readable and parseable

**Comment by villebro** [comment]
I like it!

**Comment by mistercrunch** [comment]
Here: https://issues.apache.org/jira/browse/INFRA-20236

**Comment by mistercrunch** [comment]
Leaving open until https://issues.apache.org/jira/browse/INFRA-20236 closes

**Comment by mistercrunch** [comment]
Alright I got Apache infra to install [PRLint](https://github.com/apps/prlint), with a single rule matching [Conventional
Commits](https://www.conventionalcommits.org/en/v1.0.0/) and configured it in this PR!

We can add more rules around labeling, size, assignment, body, ... Worth checking out their docs to see what's possible.

**Comment by dpgaspar** [comment]
awesome! let's merge it?

**Review by dpgaspar — APPROVED** [review]
LGTM, we can iterate over this on subsequent PR's to add feature around labels etc...

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (john-bodley) [inline .github/PULL_REQUEST_TEMPLATE.md:31]: @mistercrunch why the removal of the reviewers section?

Comment (mistercrunch) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: oh, you can just do this on the PR itself and get proper notifications related to that:
Screen Shot 2020-05-13 at 4 04 30 PM

The person opening the PR probably doesn't know who should review, so having this in the template makes outsiders feel even more that way. 

There are ways to auto-assign based on which folders/modules are touched we could use in the future.

Comment (ktmud) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: It seems only repo collaborators can request reviewers from there. Do all committers get notified when a new PR is created?

Comment (mistercrunch) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: Oh, we should add a feature to let anyone assign PRs through comments on the apache superset Github bot:
https://github.com/apache-superset/superset-github-bot

It can be a way to work around missing perms for contributors (non-committers). 

The labeling feature broke recently for reasons I could not understand (felt like unicode-related around parsing the emoji). Spent 2 hours and gave up.

**Pull request merged**
